How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,182 @@
+
+-   Set the `AWS_S3_RAW_BUCKET` environment variable to an S3 bucket where output files will be stored:
+
+```         
+import os
+import boto3
+from dotenv import load_dotenv
+```
+
+```         
+library(arrow)
+library(aws.s3)
+```
+
+-   Set the \`AWS_S3_RAW_BUCKET\` environment variable to an S3 bucket where output files will be stored:
+
+```         
+AWS_S3_RAW_BUCKET = os.environ.get("AWS_S3_RAW_BUCKET")
+```
+
+```         
+AWS_S3_RAW_BUCKET <- Sys.getenv("AWS_S3_RAW_BUCKET")
+```
+
+-   Extract the necessary data. This can be done through a few methods: scraping a webpage, utilizing an API, or downloading a file. Be aware that we may need to extract data at different time intervals. For data such as the [Central Business District](https://github.com/ccao-data/data-architecture/blob/master/etl/scripts-ccao-data-raw-us-east-1/spatial/spatial-economy.R), where the geographies are set in stone, the script can reference a geography in a single year. On the other hand, [Secondary Streets](https://github.com/ccao-data/data-architecture/blob/master/etl/scripts-ccao-data-raw-us-east-1/spatial/spatial-environment-secondary_road.R) is structured to download a different set of geographies for every year.
+
+-   Choose the output location based on your transformation strategy. If the file is already clean, and a SQL query can be easily implemented on it, the script should be stored in `ccao-data-warehouse-us-east-1`. If the file needs additional cleaning, place it in `ccao-data-raw-us-east-1`. Use a modification of the following script to upload the data to S3 in python. Note how it joins the aforementioned `AWS_S3_RAW_BUCKET`, as well as identifying the correct prefix, in this case `/housing`, before creating the unique folder `dci`.
+
+```         
+def upload_to_s3(file_content, bucket, key_prefix, file_name):
+    s3 = boto3.resource("s3")
+    object_key = f"{key_prefix}/{file_name}"
+    s3.Bucket(bucket).put_object(Key=object_key, Body=file_content)
+    return f"File uploaded to {bucket}/{object_key}"
+
+upload_to_s3(df, AWS_S3_RAW_BUCKET, "housing/dci", "dci.csv")
+```
+
+-   If you export the data to `ccao-data-warehouse-us-east-1` for direct use in a dbt model, make sure that the data is written as a parquet file. If it is written to `ccao-data-raw-us-east-1`, it can be either parquet or csv.
+-   Now that the file has been built, crawl the output with [Glue](https://us-east-1.console.aws.amazon.com/glue/home?region=us-east-1#/v2/data-catalog/crawlers) by navigating to the crawler that is configured to crawl your bucket and then selecting `Run Crawler` in the upper right.
+
+## Option B: Use a Seed to Create the Data in \`[dbtseeds](https://github.com/ccao-data/data-architecture/tree/master/dbt/seeds)\`
```

Comment:
**[Nitpick, non-blocking]** I think we are possibly missing a slash here, right? We also want to use a third-level header if we want to match the `Option A` section above.

```suggestion
### Option B: Use a Seed to Create the Data in \`[dbt/seeds](https://github.com/ccao-data/data-architecture/tree/master/dbt/seeds)\`
```

Reply:
These "\`" were not rendering well for me. Happy to add them back in if there was a purpose, but I took them out.

Reply:
Good call, I think the text looks better without the escaped backticks.